lib/pbench/server/api/resources/query_apis/datasets/datasets_contents.py

```diff
@@ -0,0 +1,207 @@
+                    "mtime": "2021-05-01T24:00:00",
+                    "mode": "0o755",
+                    "name": "1-default",
+                    "files": [
+                        {
+                            "name": "reference-result",
+                            "mtime": "2021-05-01T24:00:00",
+                            "size": 0,
+                            "mode": "0o777",
+                            "type": "sym",
+                            "linkpath": "sample1"
+                        }
+                    ],
+                    "run_data_parent": "ece030bdgfkjasdkf7435e6a7a6be804",
+                    "authorization": {
+                        "owner": "1",
+                        "access": "private"
+                    },
+                    "@timestamp": "2021-05-01T24:00:00"
+                }
+            },
+            {
+                "_index": "riya-pbench.v6.run-toc.2021-05",
+                "_type": "_doc",
+                "_id": "3bba25b62fhdgfajgsfdty6797ed06a",
+                "_score": 0.0,
+                "_source": {
+                    "parent": "/1-default",
+                    "directory": "/1-default/sample1",
+                    "mtime": "2021-05-01T24:00:00",
+                    "mode": "0o755",
+                    "name": "sample1",
+                    "ancestor_path_elements": [
+                        "1-default"
+                    ],
+                    "files": [
+                        {
+                            "name": "result.txt",
+                            "mtime": "2021-05-01T24:00:00",
+                            "size": 0,
+                            "mode": "0o644",
+                            "type": "reg"
+                        },
+                        {
+                            "name": "user-benchmark.cmd",
+                            "mtime": "2021-05-01T24:00:00",
+                            "size": 114,
+                            "mode": "0o755",
+                            "type": "reg"
+                        }
+                    ],
+                    "run_data_parent": "ece030bdgfkjasdkf7435e6a7a6be804",
+                    "authorization": {
+                        "owner": "1",
+                        "access": "private"
+                    },
+                    "@timestamp": "2021-05-01T24:00:00"
+                }
+            }
+
+        Output:
+            {
+                "directories":
+                [
+                    "sample1"
+                ],
+                "files": [
+                    {
+                        "name": "reference-result",
+                        "mtime": "2021-05-01T24:00:00",
+                        "size": 0,
+                        "mode": "0o777",
+                        "type": "sym",
+                        "linkpath": "sample1"
+                    }
+                ]
+            }
+
+        """
+        dir_list = []
```

Comment:
So, before doing anything else, we should be checking the number of "hits" and returning `NOT_FOUND` if it's zero.

(And, it would be good if the docstring mentioned that, along with what this function uses the `context` parameter for.  Also, it would be good to tweak the description of the `es_json` parameter, since the example is listing the contents of the `es_json["hits"]["hits"]` list, and that's two levels inside what the function receives.)